### DISCUSSION TITLE
Are deleted strokes useful ?

### DISCUSSION THREAD

**Discussion by mtkski** [excalidraw/excalidraw#6928] [Q&A]
Hi, when I opened an Excalidraw drawing in md (using Obsidian, I don't know if the info is relevent), I saw that all the deleted strokes were still present, they are juste flagged "isDeleted : true" in the json object. I thougth it was for undo, but once you close the file you can't undo what you did before. I also tried deleting the flagged files during editing, and I could undo. That's why I think it's actually useless. Does it have a practical use ? Does other similar software do the same ?

Because of a lack of time at the moment I didn't do a lot of testing, but I tried prompting gpt for a quick script that deletes the elements with the attribute "isDeleted : true". I found that on small test files with a few strokes the file size decreases, but when I tried on a big file (a real note from class), the size actually increased. The code seems inefficient, but as I said, I don't have time to check in details rn. 

My idea is a script that deletes the deleted strokes from the json object, thus saving space. I wanted to ask you guys so I can understand all this better for when I have time to work on this idea. Thanks a lot and sorry if it's a stupid question, I'm not used to open source, it's my first time contributing.

**Comment by hero101** [comment]
bump

**Comment by Mrazator** [comment]
We do not persist deleted elements, though Obsidian might do differently. cc @zsviczian

**Comment by zsviczian** [comment]
This is not the right thread to discuss Obsidian issues... as the example shows, I don't even notice them here.

To answer the question, yes they are useful. And there is no need to automate deletion with a script or similar as deleted strokes aren't persisted indefinitely, but just after each save, if you save twice, deleted items are removed (in other words deleted items are always removed on the second save, kind of like a rolling window of the past... The use case is collaboration in Excalidraw Obsidian using Obsidian sync. In this case at each autosave event Excalidraw needs to reconcile changes on the other instances. This is actually similar on excalidraw.com as well during collaboration... but due to technology differences this is how it works in Obsidian.
